AlphaWallet/Transfer/ViewModels/SignatureConfirmationConfirmationViewModel.swift

```diff
@@ -132,14 +132,28 @@ enum SignatureConfirmationViewModel {
     }
 
     struct EIP712TypedDataConfirmationViewModel {
-        var values: [(key: String, value: EIP712TypedData.JSON)]
+        typealias EIP712TypedDataToKey = (key: String, value: EIP712TypedData.JSON)
+        let values: [EIP712TypedDataToKey]
         let walletConnectDappRequesterViewModel: WalletConnectDappRequesterViewModel?
 
         init(data: EIP712TypedData, walletConnectDappRequesterViewModel: WalletConnectDappRequesterViewModel?) {
             self.walletConnectDappRequesterViewModel = walletConnectDappRequesterViewModel
-            values = [
-                (key: data.domainName, value: .string(data.domainVerifyingContract?.truncateMiddle ?? ""))
-            ] + data.message.keyValueRepresentationToFirstLevel
+
+            var _values: [EIP712TypedDataToKey] = []
+            
+            if data.domainName.nonEmpty {
```

Comment:
I think there should be one more check if both are nonEmpty? The original:

```
(key: data.domainName, value: .string(data.domainVerifyingContract?.truncateMiddle ?? ""))
```

Reply:
so we don't need to display domain name if there is no  `domainVerifyingContract`, right? not displaying them separatelly?

Reply:
Sorry, it's my pseudo code. How about this?

```
if notEmpty(domain.verifyigContract) && notEmpty(domain.name)
    "\(domain.name): \(domain.verifyingContract)"
elsif notEmpty(domain.verifyigContract)
    "domain.verifyingContract: \(domain.verifyingContract)"
elsif notEmpty(domain.name)
    "domain.name: \(domain.name)"
end
```